Feature/TINY-7549: Clean up Boulder ADTs, Strength concept and general things (#6894)
tinymce/tinymce: modules/boulder/CHANGELOG.md

# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

> ## Unreleased

Reviewer: We maybe should have a `Changed` here to state that we replaced ADTs?

Author: Did this, unsure about the "standard functions" wording though. Thoughts?

> ### Removed

Reviewer: `indexOnKey` was also removed from the `Objects` API.

- `FieldPresenceAdt` and `ValueProcessorAdt` (which was aliased to `FieldProcessorAdt` externally) #TINY-7549
- `asStruct` and `asStructOrDie` from ValueSchema as they were unused #TINY-7549

> - The Strength concept #TINY-7549

Reviewer 2: Same here

# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Unreleased

### Changed

> - Replaced the FieldPresence and ValueProcessor ADTs with standard functions #TINY-7549

Author: ```suggestion
- Replaced the FieldPresence and ValueProcessor ADTs with the native TypeScript equivalent #TINY-7549
```

> - Replaced the FieldPresence and ValueProcessor ADTs with the native TypeScript equivalent #TINY-7549

Reviewer 2: Should we include all the renames in here?

Author: Yeah, will do

Author: Think I got them all

# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),

> and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Author: As discussed, changelogs across the repo(s) are different, we should fix them all at once.

# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)

> and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

Reviewer 2: ```suggestion
The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```

Author: Github outdated this, so doing separately.

## Unreleased

### Changed
- Replaced `FieldPresenceAdt` with the native TypeScript equivalent, and renamed to `FieldPresence`  #TINY-7549
- Replaced `ValueProcessorAdt` (and its alias `FieldProcessorAdt`) with the native TypeScript equivalent, and renamed to `FieldProcessor` #TINY-7549

> - Renamed FieldSchema methods starting with "strict" to start with "required" for clarity #TINY-7549

Reviewer: ```suggestion
- Renamed `FieldSchema` methods starting with "strict" to start with "required" for clarity #TINY-7549
```

- Moved `anyValue`, `number`, `boolean`, `strign`, `func`, `postMessageable` from `ValueSchema` to `ValueType` #TINY-7549
- Renamed `ValueSchema` to `StructureSchema` #TINY-7549
- Renamed `Processor` type to `StructureProcessor` #TINY-7549
- Renamed `FieldSchema.state` to `FieldSchema.customField` #TINY-7549

### Removed

> - Removed `asStruct` and `asStructOrDie` from ValueSchema as they were unused #TINY-7549

Reviewer: ```suggestion
- Removed `asStruct` and `asStructOrDie` from `ValueSchema` as they were unused #TINY-7549
```

- Removed the Strength concept #TINY-7549
- Removed `indexOnKey` from the `Objects` API #TINY-7549

> - Removed Map and Set from postMessageable because not IE compatible #TINY-7549

Reviewer: ```suggestion
- Removed Map and Set from `postMessageable` because they are not IE compatible #TINY-7549
```
